thrifty-runtime/src/main/java/com/microsoft/thrifty/service/ClientBase.java

```diff
@@ -67,7 +67,11 @@ protected final Object execute(MethodCall<?> methodCall) throws Exception {
             throw new IllegalStateException("Cannot write to a closed service client");
         }
 
-        return invokeRequest(methodCall);
+        try {
+            return invokeRequest(methodCall);
+        } catch (UndeclaredServerException e) {
+            throw e.thriftException;
```

Comment:
I probably wouldn't want to expose it with the name `UndeclaredServerException`, but I can see the value in having a checked exception on synchronous client methods.  What do you think about making `ThriftException` checked?